src/main/java/org/broadinstitute/hellbender/engine/FeatureDataSource.java

```diff
@@ -210,8 +285,8 @@ public void setIntervalsForTraversal( final List<SimpleInterval> intervals ) {
         // Treat null and empty interval lists the same
         intervalsForTraversal = (intervals != null && !intervals.isEmpty()) ? intervals : null;
 
-        if ( intervalsForTraversal != null && ! hasIndex ) {
-            throw new UserException("File " + featureFile.getAbsolutePath() + " requires an index to enable traversal by intervals. " +
+        if ( intervalsForTraversal != null && ! supportsRandomAccess ) {
+            throw new UserException("File " + featureInput.getFeaturePath() + " requires an index to enable traversal by intervals. " +
                                     "Please index this file using the bundled tool " + IndexFeatureFile.class.getSimpleName());
```

Comment:
Generalize this error message: "Input x must support random access to enable traversal by intervals. If it's a file, please index it using the bundled tool IndexFeatureFile"